go/cmd/dolt/commands/ls.go

```diff
@@ -83,198 +80,142 @@ func (cmd LsCmd) Exec(ctx context.Context, commandStr string, args []string, dEn
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, lsDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
-	if apr.Contains(systemFlag) && apr.Contains(cli.AllFlag) {
-		verr := errhand.BuildDError("--%s and --%s are mutually exclusive", systemFlag, cli.AllFlag).SetPrintUsage().Build()
-		HandleVErrAndExitCode(verr, usage)
-	}
-
-	var root *doltdb.RootValue
-	var verr errhand.VerboseError
-	var label string
-	if apr.NArg() == 0 {
-		label = "working set"
-		root, verr = GetWorkingWithVErr(dEnv)
-	} else {
-		label, root, verr = getRootForCommitSpecStr(ctx, apr.Arg(0), dEnv)
+	if apr.Contains(cli.SystemFlag) && apr.Contains(cli.AllFlag) {
+		verr := errhand.BuildDError("--%s and --%s are mutually exclusive", cli.SystemFlag, cli.AllFlag).SetPrintUsage().Build()
+		return HandleVErrAndExitCode(verr, usage)
 	}
 
-	if verr == nil {
-		if !apr.Contains(systemFlag) || apr.Contains(cli.AllFlag) {
-			verr = printUserTables(ctx, root, label, apr.Contains(cli.VerboseFlag))
-			cli.Println()
-		}
-
-		if verr == nil && (apr.Contains(systemFlag) || apr.Contains(cli.AllFlag)) {
-			verr = printSystemTables(ctx, root, dEnv.DoltDB, apr.Contains(cli.VerboseFlag))
-			cli.Println()
-		}
-	}
-
-	return HandleVErrAndExitCode(verr, usage)
-}
-
-func getRootForCommitSpecStr(ctx context.Context, csStr string, dEnv *env.DoltEnv) (string, *doltdb.RootValue, errhand.VerboseError) {
-	cs, err := doltdb.NewCommitSpec(csStr)
-
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		bdr := errhand.BuildDError(`"%s" is not a validly formatted branch, or commit reference.`, csStr)
-		return "", nil, bdr.AddCause(err).Build()
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
 	}
-
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return "", nil, errhand.VerboseErrorFromError(err)
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	cm, err := dEnv.DoltDB.Resolve(ctx, cs, headRef)
-
+	query, err := constructInterpolatedDoltLsQuery(apr)
 	if err != nil {
-		return "", nil, errhand.BuildDError(`Unable to resolve "%s"`, csStr).AddCause(err).Build()
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
 	}
 
-	r, err := cm.GetRootValue(ctx)
-
+	_, _, err = queryist.Query(sqlCtx, "set @@dolt_show_system_tables = 1")
```

Comment:
Question - does it work if we just put this set call as a prefix on the main query?

"set @@dolt_show_system_tables = 1; show tables ... "

Reply:
No they don't work all in one string. I think the way `dolt sql -q` is set up doesn't allow multiple statements in one call.

Reply:
huh. Sounds like a feature we should have. ok. never mind for now!